src/sentry/dynamic_sampling/rules/biases/custom_rule_bias.py

```diff
@@ -18,17 +24,34 @@ def generate_rules(self, project: Project, base_sample_rate: float) -> List[Poly
         ret_val: List[PolymorphicRule] = []
 
         for rule in rules:
-            condition = json.loads(rule.condition)
-            ret_val.append(
-                {
-                    "samplingValue": {"type": "reservoir", "limit": rule.num_samples},
-                    "type": "transaction",
-                    "id": rule.external_rule_id,
-                    "condition": cast(Condition, condition),
-                    "timeRange": {
-                        "start": rule.start_date.strftime(CUSTOM_RULE_DATE_FORMAT),
-                        "end": rule.end_date.strftime(CUSTOM_RULE_DATE_FORMAT),
-                    },
-                }
-            )
+            try:
+                validate_rule_condition(rule.condition)
```

Comment:
Is there a reason we only do this here and not in other biases? If we need it in other places, lets move it to a reusable util fn

Reply:
The reason is that all the other biases are hardcoded, whereas this returns data from an external data source which has no guarantees of being correct and it's not static.

All the other biases have statically defined data (at most with some iterations), that can be validated in tests, which is what we do.